craft_providers/actions/snap_installer.py

```diff
@@ -96,28 +103,60 @@ def _get_host_snap_revision(snap_name: str) -> str:
     return snap_info.json()["result"]["revision"]
 
 
-def _get_target_snap_revision(snap_name: str, executor: Executor) -> Optional[str]:
+def _get_target_snap_revision_from_snapd(
+    snap_name: str, executor: Executor
+) -> Optional[str]:
     """Get the revision of the snap on the target."""
-    config = InstanceConfiguration.load(executor=executor)
-    if config is not None and config.snaps is not None:
-        snap = config.snaps.get(snap_name)
-        if snap is not None:
-            return str(snap.get("revision"))
-    return None
-
-
-def _get_store_snap_revision(snap_name: str, channel: str) -> str:
-    """Get the revision of a snap from the store."""
     quoted_name = urllib.parse.quote(snap_name, safe="")
-    url = f"http+unix://%2Frun%2Fsnapd.socket/v2/find?name={quoted_name}"
+    url = f"http://%2Frun%2Fsnapd.socket/v2/snaps/{quoted_name}"
+    cmd = ["curl", "--silent", "--unix-socket", "/run/snapd.socket", url]
```

Comment:
is the move to curl necessary? This means we will need to bundle it in our snaps (*craft) and we would need an api way to pass the full path to the binary.

Reply:
the url is also different when using curl

```
➜  ~ curl --unix-socket /run/snapd.socket 'http://%2Frun%2Fsnapd.socket/v2/snaps/snapcraft'
curl: (3) URL using bad/illegal format or missing URL
➜  ~ curl --unix-socket /run/snapd.socket http:///run/snapd.socket/v2/snaps/snapcraft      
{"type":"error","status-code":404,"status":"Not Found","result":{"message":"not found"}}%            ➜  ~ curl --unix-socket /run/snapd.socket 'http://localhost/v2/snaps/snapcraft'            
{"type":"sync","status-code":200,"status":"OK","result":{"id":"vMTKRaLjnOJQetI78HjntT37VuoyssFE","title":"snapcraft","summary":"easily create snaps","description":"Package, distribute, and update any app for Linux and IoT.\n\nSnaps are containerised software packages that are simple to create and install. They auto-update and are safe to run. And because they bundle their dependencies, they work on all major Linux systems without modification.","icon":"https://dashboard.snapcraft.io/site_media/appmedia/2018/04/Snapcraft-logo-bird.png","installed-size":65490944,"name":"snapcraft","publisher":{"id":"canonical","username":"canonical","display-name":"Canonical","validation":"verified"},"developer":"canonical","status":"active","type":"app","base":"core20","version":"7.0","channel":"7.x/candidate","tracking-channel":"7.x/candidate","ignore-validation":false,"revision":"7468","confinement":"classic","private":false,"devmode":false,"jailmode":false,"apps":[{"snap":"snapcraft","name":"snapcraft"}],"contact":"https://forum.snapcraft.io/c/snapcraft","license":"GPL-3.0","mounted-from":"/var/lib/snapd/snaps/snapcraft_7468.snap","website":"https://snapcraft.io","media":[{"type":"icon","url":"https://dashboard.snapcraft.io/site_media/appmedia/2018/04/Snapcraft-logo-bird.png","width":256,"height":256},{"type":"screenshot","url":"https://dashboard.snapcraft.io/site_media/appmedia/2019/03/shot-1.png","width":1920,"height":1080},{"type":"screenshot","url":"https://dashboard.snapcraft.io/site_media/appmedia/2019/03/banner_Gqg7Com.png","width":1218,"height":240},{"type":"screenshot","url":"https://dashboard.snapcraft.io/site_media/appmedia/2019/03/featured_HG3LAVN.png","width":1920,"height":1080},{"type":"screenshot","url":"https://dashboard.snapcraft.io/site_media/appmedia/2019/03/shot-2_nxKjIUA.png","width":1920,"height":1080}],"install-date":"2022-05-05T08:37:16.602373654-03:00"}}%           
```

Reply:
I see my concern is unfounded, curl is running on the remote not local! I will fix the localhost call though to allow curl to return properly